axum-macros/src/debug_handler.rs

```diff
@@ -61,7 +101,45 @@ impl Parse for Attrs {
 
         let body_ty = body_ty.unwrap_or_else(|| syn::parse_quote!(axum::body::Body));
 
-        Ok(Self { body_ty })
+        Ok(Self { body_ty, with_tys })
+    }
+}
+
+impl Attrs {
+    fn compute_specializations(&self) -> HashMap<syn::Ident, Vec<syn::Type>> {
```

Comment:
Never create a `HashMap` (that's iterated) in proc-macro code. Because `HashMap`s have non-deterministic iteration order, this results in multiple invocations of the same macro producing different variations (syntactically) of the same code, which leads to unnecessary cache invalidation when using incremental compilation. I think you can just replace all `HashMap`s here with `BTreeMap`s.

Reply:
i don't directly iterate over the hashmap anywhere, I always use the generic_params to iterate in order of declaration. happy to change it to a `BTreeMap` though if you prefer that. just let me know